devito/types/relational.py

```diff
@@ -208,3 +209,44 @@ def __new__(cls, lhs, rhs=0, subdomain=None, **kwargs):
 
 ops = {Ge: Lt, Gt: Le, Le: Gt, Lt: Ge}
 rev = {Ge: Le, Gt: Lt, Lt: Gt, Le: Ge}
+
+
+def relational_min(expr, s):
```

Comment:
Any chance you may reuse:

`def evalrel(func=min, input=None, assumptions=None):`

have not looks closely but maybe we can find some common way

Reply:
No `evalrel` simplifies/... a set of relations, this does not evaluate the condition to some minimal case, it finds the smallest. value for symbol `s` that satisfies `expr` in order to get a index bound